CMakeLists.txt

```diff
@@ -1,504 +1,2030 @@
+
+#[=========================================================[
+   boost dynamic libraries don't trivially support @rpath
+   linking right now (cmake's default), so just force
+   static linking for macos, or if requested on linux by flag
+#]=========================================================]
+if (static OR APPLE OR MSVC)
+  set (Boost_USE_STATIC_LIBS ON)
+endif ()
+set (Boost_USE_MULTITHREADED ON)
+if (static OR MSVC)
+  set (Boost_USE_STATIC_RUNTIME ON)
+else ()
+  set (Boost_USE_STATIC_RUNTIME OFF)
+endif ()
+find_package (Boost 1.67 REQUIRED
+  COMPONENTS
+    chrono
+    context
+    coroutine
+    date_time
+    filesystem
+    program_options
+    regex
+    serialization
+    system
+    thread)
+
+add_library (ripple_boost INTERFACE)
+add_library (Ripple::boost ALIAS ripple_boost)
+if (is_xcode)
+  target_include_directories (ripple_boost BEFORE INTERFACE ${Boost_INCLUDE_DIRS})
+  target_compile_options (ripple_boost INTERFACE --system-header-prefix="boost/")
+else ()
+  target_include_directories (ripple_boost SYSTEM INTERFACE ${Boost_INCLUDE_DIRS})
 endif()
 
-add_with_props(rippled_src_all src/ripple/unity/secp256k1.cpp
-  -I"${CMAKE_SOURCE_DIR}/"src/secp256k1
-  ${no_unused_w}
-  )
-
-foreach(cursrc
-    src/ripple/unity/beast.cpp
-    src/ripple/unity/lz4.c
-    src/ripple/unity/protobuf.cpp
-    src/ripple/unity/ripple.proto.cpp)
-
-  add_with_props(rippled_src_all ${cursrc}
-    ${rocks_db_system_header}
+target_link_libraries (ripple_boost
+  INTERFACE
+    Boost::boost
+    Boost::chrono
+    Boost::coroutine
+    Boost::date_time
+    Boost::filesystem
+    Boost::program_options
+    Boost::regex
+    Boost::serialization
+    Boost::system
+    Boost::thread)
+
+#[===================================================================[
+   NIH dep: openssl
+#]===================================================================]
+
+if (APPLE AND NOT DEFINED ENV{OPENSSL_ROOT_DIR})
+  find_program (HOMEBREW brew)
+  if (NOT HOMEBREW STREQUAL "HOMEBREW-NOTFOUND")
+    execute_process (COMMAND ${HOMEBREW} --prefix openssl
+      OUTPUT_VARIABLE OPENSSL_ROOT_DIR
+      OUTPUT_STRIP_TRAILING_WHITESPACE)
+  endif ()
+endif ()
+
+if ((NOT DEFINED OPENSSL_ROOT) AND (DEFINED ENV{OPENSSL_ROOT}))
+  set (OPENSSL_ROOT $ENV{OPENSSL_ROOT})
+endif ()
+file (TO_CMAKE_PATH "${OPENSSL_ROOT}" OPENSSL_ROOT)
+
+if (static OR APPLE OR MSVC)
+  set (OPENSSL_USE_STATIC_LIBS ON)
+endif ()
+set (OPENSSL_MSVC_STATIC_RT ON)
+set (_ssl_min_ver 1.0.2)
+# HACK for travis
+if ("$ENV{CI}" STREQUAL "true" AND "$ENV{TRAVIS}" STREQUAL "true")
+  set (_ssl_min_ver 1.0.1)
+endif ()
+find_package (OpenSSL ${_ssl_min_ver} REQUIRED)
+target_link_libraries (ripple_libs
+  INTERFACE
+    OpenSSL::SSL
+    OpenSSL::Crypto)
+#[=========================================================[
+   depending on how openssl is built, it might depend
+   on zlib. In fact, the openssl find package should
+   figure this out for us, but it does not currently...
```

Comment:
Did you file or look for an issue upstream with Kitware in adding support for this?

Reply:
yes: https://gitlab.kitware.com/cmake/cmake/issues/16885  ...I can add a comment linking to that issue